presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TableLayoutRewriter.java

```diff
@@ -80,12 +84,12 @@ public PlanNode planTableScan(TableScanNode node, Expression predicate)
                 .filter(layoutHasAllNeededOutputs(node))
                 .collect(toImmutableList());
         if (layouts.isEmpty()) {
-            return new ValuesNode(idAllocator.getNextId(), node.getOutputSymbols(), ImmutableList.of());
```

Comment:
Shouldn't you make `node.getCurrentConstrait()` part of a layout query? What about following scenario:
```
FilterNode(a IN (1, 2))
 TableScanNode(currentConstrait=none)
```
then first `PickTableLayout` rule would produce
```
FilterNode(a IN (2))
 TableScanNode(currentConstrait=(a IN (1))
```
but the 3rd call of `PickTableLayout` rule would produce:
```
ValuesNode(rows=[])
```
since intersect here: https://github.com/prestodb/presto/pull/8735/commits/0cc4b665b05afed5e08f24b44d037268d7c16dbc#diff-80422a11a5eab4c4e0e29dc080660e84R77 would produce empty `TupleDomain`.

Reply:
It would be incorrect to push down `a in (1)` to the tablescan.  `a in(1,2)` is basically an or `a = 1 OR a =2`.  So you couldn't filter by just one of them.